### PR TITLE
feat: add docs for native sign in using Apple SDK

### PR DESCRIPTION
**Pull request by jonas-jonas**
## Related Issue or Design Document

https://github.com/ory/kratos/pull/3476

## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added the necessary documentation within the code base (if appropriate).

### CODE REVIEW

**Comment by jonas-jonas** [comment]
Pls don't merge yet, not on Ory Network yet.

Google is not implemented yet, will do next!

File: docs/kratos/social-signin/30_apple.mdx [hunk 1]
```diff
@@ -178,6 +178,42 @@ Follow these steps to add Apple as a social sign-in provider to your project usi
 </Tabs>
 ````
 
+## Using the Apple SDK on native apps
+
+Apple provides a more integrated UX for native apps using the
+[Apple SDK](https://developer.apple.com/documentation/sign_in_with_apple/implementing_user_authentication_with_sign_in_with_apple).
+Since this flow does not involve a browser and redirect URL, but instead issues an `id_token`, it requires a bit more work to
+integrate with Ory.
```
Comment (Benehiko) [inline docs/kratos/social-signin/30_apple.mdx:186]: Always positive! :)

```suggestion
This flow uses the native Apple SDK and does not require a browser. This results in an issued `id_token`  on the client side which is exchanged at Ory for a session token.
```

Comment (Benehiko) [reply docs/kratos/social-signin/30_apple.mdx:186]: Move the limitations and drawbacks to a section at the end.

File: docs/kratos/social-signin/30_apple.mdx [hunk 2]
```diff
@@ -178,6 +178,42 @@ Follow these steps to add Apple as a social sign-in provider to your project usi
...
+
+The following steps are required to integrate the Apple SDK with Ory:
+
+1. Configure an Apple social sign-in provider in Ory using the same `client_id` as in your native app.
+1. Generate a random value that you can use as a `nonce`.
+1. Obtain an `id_token` from Apple using the Apple SDK. Make sure to also submit the SHA256 encrypted `nonce`.
+1. Submit the `id_token` and `nonce` (as the `raw_id_token_nonce`) as part of the `updateRegistrationFlow` or `updateLoginFlow`
+   request to Ory.
+1. Ory will validate the `id_token` and create an identity and optionally a session (if configured).
```
Comment (Benehiko) [inline docs/kratos/social-signin/30_apple.mdx:195]:
```suggestion
1. Configure an Apple social sign-in provider in Ory using the same `client_id` as in your native app.
2. Generate a random value that you can use as a `nonce`.
3. Obtain an `id_token` from Apple using the Apple SDK. Make sure to also submit the SHA256 encrypted `nonce`.
4. Submit the `id_token` and `nonce` (as the `raw_id_token_nonce`) as part of the `updateRegistrationFlow` or `updateLoginFlow`
   request to Ory.
5. Ory will validate the `id_token` and create an identity and optionally a session (if configured).
```

File: docs/kratos/social-signin/30_apple.mdx [hunk 3]
```diff
@@ -178,6 +178,42 @@ Follow these steps to add Apple as a social sign-in provider to your project usi
...
+
+The `id_token` is verified using Apple's publicly available signing keys, available under https://appleid.apple.com/auth/keys.
```
Comment (Benehiko) [inline docs/kratos/social-signin/30_apple.mdx:197]:
```suggestion
The `id_token` is verified using Apple's publicly available signing keys. For more information, please take a look at the [Apple documentation](https://appleid.apple.com/auth/keys).
```

**Review by Benehiko — CHANGES_REQUESTED** [review]
Could you also link the documentation to here from https://docs-cazi73gk0-ory.vercel.app/docs/kratos/social-signin/native-apps? Since it is now possible to have complete native flows instead of browser in native apps.

File: docs/kratos/social-signin/30_apple.mdx (hunk 1 above)
Comment (Benehiko) [reply docs/kratos/social-signin/30_apple.mdx]: no i meant that the drawbacks shouldn't be listed within the highlevel description.

**Review by Benehiko — APPROVED** [review]

**Pull request merged**